vochain/state.go

```diff
@@ -634,14 +634,15 @@ func (v *State) Envelope(processID, nullifier []byte, isQuery bool) (_ []byte, e
 
 // EnvelopeExists returns true if the envelope identified with voteID exists
 func (v *State) EnvelopeExists(processID, nullifier []byte, isQuery bool) (bool, error) {
-	e, err := v.Envelope(processID, nullifier, isQuery)
-	if err != nil && err != ErrVoteDoesNotExist {
-		return false, err
-	}
-	if err == ErrVoteDoesNotExist {
+	_, err := v.Envelope(processID, nullifier, isQuery)
+	if err == ErrProcessNotFound {
+		return false, nil
+	} else if err == ErrVoteDoesNotExist {
```

Comment:
`errors.Is` for these too; this helps ensure that future error wrapping won't break our code.